extensions/amp-accordion/1.0/README.md

```diff
@@ -0,0 +1,641 @@
+
+```javascript
+ref.current.collapse();
+ref.current.collapse('section1');
+```
+
+##### Events
+
+The Bento Accordion API allows you to respond to the following events:
+
+**onExpandStateChange**
+This event is triggered on a section when an accordion section is expanded or collpased and is dispatched from the expanded section.
+
+See below for example.
+
+**onCollapse**
+This event is triggered on a section when an accordion section is collapsed and is dispatched from the collapsed section.
+
+In the example below, `section 1` listens for the `expand` event and expands `section 2` when it is expanded. `section 2` listens for the `collapse` event and collapses `section 1` when it is collapsed.
+
+See below for example.
+
+```jsx
+    <BentoAccordion ref={ref}>
+      <BentoAccordionSection
+        id="section1"
+        key={1}
+        onExpandStateChange={
+          (expanded) => expanded ? alert('section1 expanded') : alert('section1 collapsed')
+        }>
+        <BentoAccordionHeader><h1>Section 1</h1></BentoAccordionHeader>
+        <BentoAccordionContent>Content 1</BentoAccordionContent>
+      </BentoAccordionSection>
+
+      <BentoAccordionSection
+        id="section2"
+        key={2}
+        onExpandStateChange={
+          (expanded) => expanded ? alert('section2 expanded') : alert('section2 collapsed'}
+        }>
+        <BentoAccordionHeader><h1>Section 2</h1></BentoAccordionHeader>
+        <BentoAccordionContent>Content 2</BentoAccordionContent>
+      </BentoAccordionSection>
+
+      <BentoAccordionSection
+        id="section3"
+        key={3}
+        onExpandStateChange={
+          (expanded) => expanded ? alert('section3 expanded') : alert('section3 collapsed'}
+        }>
+        <BentoAccordionHeader><h1>Section 3</h1></BentoAccordionHeader>
+        <BentoAccordionContent>Content 3</BentoAccordionContent>
+      </BentoAccordionSection>
+    </BentoAccordion>
+```
+
+#### Layout and style
+
+**Container type**
+
+The `BentoAccordion` component has a defined layout size type. To ensure the component renders correctly, be sure to apply a size to the component and its immediate children via a desired CSS layout (such as one defined with `height`, `width`, `aspect-ratio`, or other such properties). These can be applied inline:
+
+```jsx
+<BentoAccordion style={{width: '300px', height: '100px'}}>
+  ...
+</BentoAccordion>
+```
+
+Or via `className`:
+
+```jsx
+<BentoAccordion className='custom-styles'>
+  ...
+</BentoAccordion>
+```
+
+```css
+.custom-styles {
+  height: 100px;
+  width: 100%;
+}
+```
+
+#### Props
+
+##### BentoAccordion
+
+###### animate
+
+If true, then uses "roll-down" / "roll-up" animation during the expansion and collapse of each section
+Default: `false`
+
+###### as
+
+todo
+
+###### expandSingleSection
+
+If true, then expanding 1 section will automatically collapse all other sections:
+Default: `false`
+
+###### id
+
+todo
```

Comment:
`as`/`class`/`id`/`role`/`tabindex` may be better formatted as common supported props in some central Preact/React component documentation file and referenced here, similar to https://amp.dev/documentation/guides-and-tutorials/learn/common_attributes/ in AMP mode.

Reply:
where would recommend I create that file? `extensions/README.md`?

Reply:
I'd say `docs/spec/bento-common-props.md`

Reply:
But feel free to do that in a separate PR

Reply:
moved `docs/spec/bento-common-props.md`